### ISSUE TITLE
Extended csmith for gcc C language extensions

### ISSUE THREAD

**Issue by Sameeranjoshi** [csmith-project/csmith#60]
Hi,We have implemented following gcc c extensions in csmith. Is it
 possible to take consideration of these implemented extensions in csmith next  release?
Thanks.

**Comment by regehr** [comment]
The best way to accomplish this goal is to give us a series of smallish pull requests, each of which makes sense on its own. I'm not sure that we're going to be able to review / accept a huge change.

**Comment by regehr** [comment]
also, are the results of this work written up anywhere? how many bugs were found using it, and in what compilers?

**Comment by shubhamnarlawar77** [comment]
We had reported 1 issue found when we added Tm-relaxed extension

Link of gcc bugszilla
(https://gcc.gnu.org/bugzilla/show_bug.cgi?id=87118)


Using the extended csmith for gcc c language extensions.

We are currently working on adding more extensions to it and fuzzing in background.

**Comment by shubhamnarlawar77** [comment]
We have successfully implemented following Gcc C extensions.


1. Computed goto

2. Statement expression

3. Typeof

4. Signed and unsigned int128

5. Local labels

6. Binary constants

7. Union cast

8. Variable attributes - aligned, packed, unused, section

9. Label attributes - hot cold

10. Tm-relaxed

**Comment by Sameeranjoshi** [comment]
> also, are the results of this work written up anywhere? how many bugs were found using it, and in what compilers?

No,we have been working under the guidance of our mentor from gcc (Andi Kleen), where we initially aimed to create something working for GCC and try fuzzing it in background. Currently we are working on finding the code coverage of extension which have been implemented so as to announce to gcc community.

**Comment by regehr** [comment]
sounds great.

Again, the thing that makes it easiest to accept your code is some smaller, self-contained pull requests. Do you think that the patch can be broken up in a sensible way? If not, you can try submitting a giant pull request but in that case it will take us much longer to review it.

**Comment by shubhamnarlawar77** [comment]
Hi, I have send you pull request containing implementation of computed goto.

Command line options for computed goto are as below -

   1. --computed-goto : Enables computed goto extension
   2. --no-computed-goto : Disbales computed goto exxtension

By default, this extension is disabled.


Please review and suggests changes if necessary.

Thanks. Regards

Shubham

**Comment by jxyang** [comment]
Hi  Shubham,

Thanks for the patch. I have made a few comments with your change. Look
forward to your other pull requests.

Thanks,

-Xuejun

**Comment by jxyang** [comment]
Sent from my iPhone

> On Dec 21, 2018, at 7:29 AM, Shubham Narlawar <notifications@github.com> wrote:
> 
> Yeah, sure.
> 
> On Fri, Dec 21, 2018 at 10:42 PM Xuejun Yang <notifications@github.com>
> wrote:
> 
> > Hi Shubham,
> >
> > Thanks for the patch. I have made a few comments with your change. Look
> > forward to your other pull requests.
> >
> I did not get this line, few comments with your change. Can you please
> elaborate?
> 
I reviewed your pull request. Let’s continue the discussion in that thread.

**Comment by shubhamnarlawar77** [comment]
On Sat, Dec 22, 2018 at 6:42 AM Xuejun Yang <notifications@github.com>
wrote:

>
>
> Sent from my iPhone
>
> > On Dec 21, 2018, at 7:29 AM, Shubham Narlawar <notifications@github.com>
> wrote:
> >
> > Yeah, sure.
> >
> > On Fri, Dec 21, 2018 at 10:42 PM Xuejun Yang <notifications@github.com>
> > wrote:
> >
> > > Hi Shubham,
> > >
> > > Thanks for the patch. I have made a few comments with your change. Look
> > > forward to your other pull requests.
> > >
> > I did not get this line, few comments with your change. Can you please
> > elaborate?
> >
> I reviewed your pull request. Let’s continue the discussion in that thread.
>
The build for this pull request exited with 0. I am unable to see
discussion in that thread. What's the further process for merging this pull
request?
